app/assets/src/components/utils/report.js

```diff
@@ -0,0 +1,140 @@
+    resultTaxons.push(genusTaxon);
+    resultTaxons = resultTaxons.concat(matchedTaxons);
+  } else if (
+    ThresholdMap.taxonPassThresholdFilter(genusTaxon, activeThresholds)
+  ) {
+    resultTaxons.push(genusTaxon);
+  }
+
+  return resultTaxons;
+};
+
+export const isTaxonIncluded = (
+  taxon,
+  includedCategories,
+  includedSubcategoryColumns,
+  excludedSubcategoryColumns
+) => {
+  // returns if taxon is in either the included categories / subcategories AND
+  // the taxon is not in an excluded subcategory
+  return (
+    (includedCategories.indexOf(taxon.category_name) >= 0 ||
+      includedSubcategoryColumns.some(column => {
+        return taxon[column] == 1;
+      })) &&
+    !excludedSubcategoryColumns.some(column => {
+      return taxon[column] == 1;
+    })
+  );
+};
+
+export const getTaxonSortComparator = (
+  primarySortParams,
+  secondarySortParams,
+  genusMap
+) => {
+  const [ptype, pmetric] = primarySortParams;
+  const [stype, smetric] = secondarySortParams;
+
+  return (a, b) => {
+    const genus_a = genusMap[a.genus_taxid];
+    const genus_b = genusMap[b.genus_taxid];
+
+    const genus_a_p_val = parseFloat(genus_a[ptype][pmetric]);
+    const genus_a_s_val = parseFloat(genus_a[stype][smetric]);
+    const a_p_val = parseFloat(a[ptype][pmetric]);
+    const a_s_val = parseFloat(a[stype][smetric]);
+
+    const genus_b_p_val = parseFloat(genus_b[ptype][pmetric]);
+    const genus_b_s_val = parseFloat(genus_b[stype][smetric]);
+    const b_p_val = parseFloat(b[ptype][pmetric]);
+    const b_s_val = parseFloat(b[stype][smetric]);
+
+    // compared at genus level descending and then species level descending
+    if (a.genus_taxid == b.genus_taxid) {
+      // same genus
+      if (a.tax_level > b.tax_level) {
+        return -1;
+      } else if (a.tax_level < b.tax_level) {
+        return 1;
+      }
+      if (a_p_val > b_p_val) {
+        return -1;
+      } else if (a_p_val < b_p_val) {
+        return 1;
+      }
+      if (a_s_val > b_s_val) {
+        return -1;
+      } else if (a_s_val < b_s_val) {
+        return 1;
+      }
+      return 0;
+    }
+
+    if (genus_a_p_val > genus_b_p_val) {
+      return -1;
+    } else if (genus_a_p_val < genus_b_p_val) {
+      return 1;
+    }
+
+    if (genus_a_s_val > genus_b_s_val) {
+      return -1;
+    } else if (genus_a_s_val < genus_b_s_val) {
+      return 1;
+    }
+
+    if (a.genus_taxid < b.genus_taxid) {
+      return -1;
+    } else if (a.genus_taxid > b.genus_taxid) {
+      return 1;
+    }
+  };
+};
+
+export const getCategoryAdjective = category => {
+  const category_lowercase = category.toLowerCase();
+  switch (category_lowercase) {
+    case "bacteria":
+      return "bacterial";
+    case "archaea":
+      return "archaeal";
+    case "eukaryota":
+      return "eukaryotic";
+    case "viruses":
+      return "viral";
+    case "viroids":
+      return "viroidal";
+    case "uncategorized":
+      return "uncategorized";
```

Comment:
Add a default case: You can actually return the category name by default and you do not need the last case.